src/Sarif.UnitTests/Core/FileDataTests.cs

```diff
@@ -0,0 +1,117 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Security.AccessControl;
+using System.Text;
+using FluentAssertions;
+
+using Microsoft.CodeAnalysis.Sarif.Writers;
+
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Sarif
+{
+    public class FileDataTests
+    {
+        [Fact]
+        public void FileData_Create_NullUri()
+        {
+            Action fileDataCreation = () => { FileData.Create(null, LoggingOptions.None); };
+
+            fileDataCreation.ShouldThrow<ArgumentNullException>();
+        }
+
+        [Fact]
+        public void FileData_ComputeHashes()
+        {
+            string filePath = Path.GetTempFileName();
+            string fileContents = Guid.NewGuid().ToString();
+            Uri uri = new Uri(filePath);
+
+            try
+            {
+                File.WriteAllText(filePath, fileContents);
+                FileData fileData = FileData.Create(uri, LoggingOptions.ComputeFileHashes);
+                fileData.Uri.Should().Be(uri.ToString());
```

Comment:
>; [](start = 56, length = 1)

For symmetry with next test, assert that the file contents were _not_ persisted.